bots/machine/machine_core/machine_base.py

```diff
@@ -0,0 +1,276 @@
+# -*- coding: utf-8 -*-
+
+# This file is part of Cockpit.
+#
+# Copyright (C) 2013 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+import os
+
+from . import ssh_connection
+from . import timeout
+from .constants import DEFAULT_IDENTITY_FILE, ATOMIC_IMAGES, TEST_DIR
+
+LOGIN_MESSAGE = """
+TTY LOGIN
+  User: {ssh_user}/admin  Password: foobar
+  To quit use Ctrl+], Ctrl+5 (depending on locale)
+
+SSH ACCESS
+  $ ssh -p {ssh_port} {ssh_user}@{ssh_address}
+  Password: foobar
+
+COCKPIT
+  http://{web_address}:{web_port}
+"""
+
+RESOLV_SCRIPT = """
+set -e
+# HACK: Racing with operating systems reading/updating resolv.conf and
+# the fact that resolv.conf can be a symbolic link. Avoid failures like:
+# chattr: Operation not supported while reading flags on /etc/resolv.conf
+mkdir -p /etc/NetworkManager/conf.d
+printf '[main]\ndns=none\n' > /etc/NetworkManager/conf.d/dns.conf
+systemctl reload-or-restart NetworkManager
+printf 'domain {domain}\nsearch {domain}\nnameserver {nameserver}\n' >/etc/resolv2.conf
+chcon -v unconfined_u:object_r:net_conf_t:s0 /etc/resolv2.conf 2> /dev/null || true
+mv /etc/resolv2.conf /etc/resolv.conf
+"""
+
+
+class Machine(ssh_connection.SSHConnection):
```

Comment:
The main class here is `Machine`, so the module should also be called `machine.py`, not `machine_base.py`. Let's try to match up the file names to the class names.